### ISSUE TITLE
Suggested policy: duplicate use and require

### ISSUE THREAD

**Issue by epa** [Perl-Critic/Perl-Critic#928]
A new policy could detect exact duplicate `use` statements in the same package.  These two examples would give a message:

```
use Foo;
use Bar;
use Foo;
```

```
use Foo qw(frob);
say 'hi there';
use Foo 'frob';
```

Yes, there are many other ways for something to be loaded or imported twice: what if the import list is different in the two cases?  But for this suggested policy I am only considering the case when the two lines give the same module name and the same import list, after parsing.

As a further refinement, perlcritic could warn on `require` when the same module has already been `use`d.

**Comment by petdance** [comment]
What problems would this be solving?

If I have

    use Foo;
    ....
    require Foo;

is there a problem?  The `require` is unnecessary, certainly.  Anything else?

**Comment by epa** [comment]
Yes, that's all.  Code that's at best unnecessary, and may indicate some confusion on the part of the programmer.  (I thought I was saving startup time by only requiring `Foo` when it was needed, not realizing that `use Foo` had crept back into the file meanwhile.)

`require` in itself is a little bit unusual; in my view the only reason to use it is if you particularly want to defer module loading until run time -- and that's pointless if you have already loaded it at compile time with a `use` statement.

**Comment by petdance** [comment]
I'm certainly not against identifying unnecessary code.  Just clarifying that there wasn't some other problem this could introduce.

I'm even more interested in the scenario of two different `use`s with different import lists.

**Comment by epa** [comment]
I guess you're right, any kind of `use` of the same module twice in the same package is worth flagging.  If the import lists are the same: just redundant(*).  If they differ: something odd is going on.

(*) Except for bizarre modules that perform some side effect based on their import list.  But no need to worry about those I think.
